sdk/metricsadvisor/Azure.AI.MetricsAdvisor/perf/Tests/GetAnomaliesForAlertTest.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Azure.AI.MetricsAdvisor.Perf
+{
+    public class GetAnomaliesForAlertTest : MetricsAdvisorTest<MetricsAdvisorPerfOptions>
+    {
+        public GetAnomaliesForAlertTest(MetricsAdvisorPerfOptions options) : base(options)
+        {
+            ValidateOptions();
+        }
+
+        public override void Run(CancellationToken cancellationToken)
+        {
+            int count = 0;
+
+            foreach (var _ in Client.GetAnomalies(AlertConfigurationId, AlertId, cancellationToken: cancellationToken))
+            {
+                if (++count >= Options.Count)
+                {
+                    break;
+                }
+            }
+        }
+
+        public override async Task RunAsync(CancellationToken cancellationToken)
+        {
+            int count = 0;
+
+            await foreach (var _ in Client.GetAnomaliesAsync(AlertConfigurationId, AlertId, cancellationToken: cancellationToken))
+            {
+                if (++count >= Options.Count)
+                {
+                    break;
+                }
+            }
+        }
+
+        private void ValidateOptions()
```

Comment:
We can set the number of anomalies we'll list in our test with the `--count` command line parameter. If this number is greater than the actual amount of anomalies we have available in the Azure resource, we'll finish listing them sooner than expected during the test. This can end up reducing the measured time and giving us a misleading result. I've added an extra check to be run before setup just to make sure we have at least the amount of anomalies we are expecting to get from the service. I'm doing the same for every test.